<a href="https://colab.research.google.com/github/sureshbudha879-pixel/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

I selected the **Content Refresh Prioritization** lane.

My goal is to identify content pages that are most likely to benefit from updates before their search performance declines further. Since organizations manage many pages, reviewing every page manually is difficult and time-consuming. This lane focuses on using historical search performance and content characteristics to prioritize which pages should be reviewed first. I chose this lane because it directly supports SEO and content teams in making data-informed decisions.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The Question

### Decision
Which content pages should be prioritized for review and content refresh?

### Action
The SEO or content team can use these recommendations to decide which pages should be updated first to improve or maintain search performance.

### Cost of a Wrong Recommendation
If important pages are not recommended, they may continue losing search visibility and traffic.

If pages are incorrectly recommended, the team may spend time and resources updating content that did not require changes.

### Why Machine Learning Can Help
Machine learning can analyze multiple page characteristics, such as impressions, CTR, search position, and content age, to identify useful patterns. The goal is to support decision-making rather than replace human judgment.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))

print("\nTrend Direction")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(),3))

print("Average Search Position:", round(df["avg_position"].mean(),2))

Number of pages: 30000

Trend Direction
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR: 0.511
Average Search Position: 16.34


## Quick Look at the Data

The starter dataset contains 30,000 content pages. Among them, 16,262 pages (over half of the dataset) show a downward trend, indicating that many pages may require review or content refresh. The average CTR is 0.511, and the average search position is 16.34, showing meaningful variation in page performance. These observations suggest that the dataset is suitable for building a decision-support model for content refresh prioritization.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project provides decision support based on patterns observed in the available data.

The analysis identifies directional relationships that may help prioritize content for review. It does not prove causation or predict Google's ranking algorithm.

Any recommendations produced by this project should be treated as evidence-based guidance rather than guaranteed outcomes.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.